### PR TITLE
Launch simulator after creating a ScreenBuilder project

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Includes some refactoring (introduction of a new service) , a minor bug fix (caching of the project directory) and an option to prevent simulator from launching
Implements [this story](http://teampulse.telerik.com/view#item/293551)
Merge after https://github.com/telerik/mobile-cli-lib/pull/365

### CODE REVIEW

File: lib/project.ts [hunk 1]
```diff
@@ -171,7 +171,7 @@ export class Project implements Project.IProject {
 			if(this.cachedProjectDir !== "") {
```
Comment (rosen-vladimirov) [inline lib/project.ts]: shouldn't you change this line to 
```TypeScript
if(this.cachedProjectDir) {
```

File: lib/services/simulatorPlatformServices.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import path = require("path");
+import Future = require("fibers/future");
```
Comment (teobugslayer) [inline lib/services/simulatorPlatformServices.ts]: Are you using this anywhere?

File: lib/services/simulatorPlatformServices.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+class WinSimulatorPlatformServices implements IExtensionPlatformServices {
+	private static PACKAGE_NAME_WIN: string = "Telerik.BlackDragon.Client.Mobile.Tools.Package";
+	private static EXECUTABLE_NAME_WIN = "Icenium.Simulator.exe";
+
+	constructor(private $childProcess: IChildProcess,
+				private $hostInfo: IHostInfo) { }
+
+	public getPackageName(): string {
```
Comment (teobugslayer) [inline lib/services/simulatorPlatformServices.ts]: Maybe we can make this a property as well?

File: lib/services/simulatorPlatformServices.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		this.$childProcess.spawn('open', commandLine,
+			{ stdio:  ["ignore", "ignore", "ignore"], detached: true }).unref();
+	}
+
+	public canRunApplication(): IFuture<boolean> {
+		return (() => true).future<boolean>()();
+	}
+}
+
+let hostInfo = $injector.resolve("hostInfo");
```
Comment (teobugslayer) [inline lib/services/simulatorPlatformServices.ts]: you may give a type to hostInfo

**Comment by Mitko-Kerezov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
Ping @rosen-vladimirov @teobugslayer @Fatme for the code and @ikoevska for help with the new option

File: docs/man_pages/project/creation/create-screenbuilder.md [hunk 5]
```diff
@@ -3,13 +3,14 @@ create screenbuilder
...
 ------|-------
-General | `$ appbuilder create screenbuilder <App name> [--path <Directory>] [--appid <App ID>]`
+General | `$ appbuilder create screenbuilder <App name> [--path <Directory>] [--appid <App ID>] [--no-simulator]`
 
 Creates a new project for hybrid development with Screen Builder. <% if(isHtml) { %>Screen Builder lets you create a new project customized with navigation, home view and user interface skin. Later you can run the Screen Builder commands on this project to further customize it by adding application views, connecting to data sources, creating user registration and sign-in and adding forms, lists and fields.<% } %> 
 
 ### Options
 * `--path` - Specifies the directory where you want to create the project, if different from the current directory. The directory must be empty.
 * `--appid` - Sets the application identifier for your app. If not specified, the application identifier is set to `com.telerik.<App name>`. 
+* `--no-simulator` - Specifies that the device simulator should not be launched after successful project creation.
```
Comment (rosen-vladimirov) [inline docs/man_pages/project/creation/create-screenbuilder.md]: awesome

**Comment by rosen-vladimirov** [comment]
:+1:  after green build

**Comment by Mitko-Kerezov** [comment]
run ci

File: docs/man_pages/project/creation/create-screenbuilder.md (hunk 5 above)
Comment (ikoevska) [inline docs/man_pages/project/creation/create-screenbuilder.md]: If set, the Telerik AppBuilder CLI does not launch he device simulator after the project is created successfully.

**Comment by Mitko-Kerezov** [comment]
run ci

**Pull request merged**